bsp/Src/CAN.c

```diff
@@ -765,8 +765,10 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan) {
 }
 
 // CAN1
+#ifdef CAN1
```

Comment:
Won't the check in the header cause an error anyways? Is this not redundant?

Reply:
u right

Reply:
no wait this was a problem of the HAL_CAN_IRQHandler method not being deffed in but the file still being compiled. So if I try to compile code for a mcu that doesn't have CAN but I still want to test on, the HAL_CAN_IRQ function won't exist and I'll get an error during compilation since technically  CAN is still being compiled; even though I may not be using the CAN header at all.

Reply:
you can see a similar thing is already implemented on UART to prevent that